lib/app/features/chat/providers/user_chat_relays_provider.c.dart

```diff
@@ -37,3 +41,37 @@ Future<UserChatRelaysEntity?> userChatRelays(
       .watch(nostrNotifierProvider.notifier)
       .requestEntity<UserChatRelaysEntity>(requestMessage, actionSource: ActionSourceUser(pubkey));
 }
+
+///
+/// This provider is used to set the user chat relays for a given pubkey.
+/// If the user chat relays are already set, it will not set them again.
+/// If the user's relays have been updated, it will update the chat relays to match.
+///
+@riverpod
+Future<void> setUserChatRelays(Ref ref) async {
+  final pubkey = ref.watch(currentPubkeySelectorProvider);
+
+  if (pubkey == null) {
+    throw UserMasterPubkeyNotFoundException();
+  }
+  final userChatRelays = await ref.watch(userChatRelaysProvider(pubkey).future);
+
+  final userRelays = await ref.watch(userRelaysManagerProvider.notifier).fetch([pubkey]);
+  final relayUrls = userRelays.first.data.list.map((e) => e.url).toList();
+
+  if (userChatRelays != null) {
+    final chatRelays = userChatRelays.data.list.map((e) => e.url).toList();
+    if (chatRelays == relayUrls) {
+      return;
+    }
+  }
+
+  final chatRelays = UserChatRelaysData(
+    list: relayUrls.map((url) => UserRelay(url: url)).toList(),
+  );
+
+  final chatRelaysEvent = await ref.read(nostrNotifierProvider.notifier).sign(chatRelays);
+
+  await ref.read(nostrNotifierProvider.notifier).sendEvents([chatRelaysEvent]);
```

Comment:
Avoid side effects in a provider's `build`

Reply:
fixed